muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/SyncStudentEventHandler.java

```diff
@@ -53,8 +74,40 @@ public synchronized void onSchoolDataWorkspaceUserRemovedEvent(@Observes SchoolD
     
     WorkspaceEntity workspace = workspaceEntityController.findWorkspaceByDataSourceAndIdentifier(workspaceDataSource, workspaceIdentifier);
     
-    
     chatSyncController.removeChatRoomMembership(user, workspace);
     
   }
+  
+  public synchronized void onWorkspaceChatSettingsEnabledEvent (@Observes WorkspaceChatSettingsEnabledEvent event) {
+    //jos kurssin chatti pistetään päälle niin kaikki sen opiskelijat läpi ja ne, joilla on chatti päällä niin mukaan
+
+    String workspaceIdentifier = event.getWorkspaceIdentifier();
+    String workspaceDataSource = event.getWorkspaceDataSource();
+    WorkspaceEntity workspaceEntity = workspaceEntityController.findWorkspaceByDataSourceAndIdentifier(workspaceDataSource, workspaceIdentifier);
+
+    
+    List<WorkspaceUserEntity> activeWorkspaceStudents = workspaceUserEntityController.listActiveWorkspaceStudents(workspaceEntity);
+    List<WorkspaceUserEntity> workspaceStaffMembers = workspaceUserEntityController.listActiveWorkspaceStaffMembers(workspaceEntity);
+    for (WorkspaceUserEntity activeWorkspaceStudent : activeWorkspaceStudents) {
+      SchoolDataIdentifier user = new SchoolDataIdentifier(activeWorkspaceStudent.getUserSchoolDataIdentifier().getIdentifier(), activeWorkspaceStudent.getUserSchoolDataIdentifier().getDataSource().getIdentifier());
+      
+      UserChatSettings userChatSetting = chatController.findUserChatSettings(user);
+      
+      if (userChatSetting != null && userChatSetting.getVisibility().toString() == "VISIBLE_TO_ALL"){
+        chatSyncController.syncStudent(user);
+      }
+      
+    }
+    
+    for (WorkspaceUserEntity workspaceStaffMember : workspaceStaffMembers) {
+
+      SchoolDataIdentifier user = new SchoolDataIdentifier(workspaceStaffMember.getUserSchoolDataIdentifier().getIdentifier(), workspaceStaffMember.getUserSchoolDataIdentifier().getDataSource().getIdentifier());
+      UserChatSettings userChatSetting = chatController.findUserChatSettings(user);
+      
+      if (userChatSetting != null && userChatSetting.getVisibility().toString() == "VISIBLE_TO_ALL"){
```

Comment:
Don't use .toString with enums where possible, better to check with 
`UserChatVisibility.VISIBLE_TO_ALL.equals(userChatSettings.getVisibility())` keep the null check though.